extensions/tasks/NLUTestV0/publishTestResults.ts

```diff
@@ -0,0 +1,50 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+
+type TestRunner = "JUnit" | "NUnit" | "VSTest" | "XUnit" | "CTest";
+
+const TESTRUN_SYSTEM = 'VSTS - PTR';
+
+function isNullOrWhitespace(input: any) {
+    if (typeof input === 'undefined' || input === null) {
+        return true;
+    }
+    return input.replace(/\s/g, '').length < 1;
+}
+
+function publish(testRunner: TestRunner, resultFiles: string[], testRunSystem: string) {
+    var properties: { [key: string]: string | string[] } = {};
+    properties['type'] = testRunner;
+
+    if (resultFiles) {
+        properties['resultFiles'] = resultFiles;
+    }
+    properties['testRunSystem'] = testRunSystem;
+
+    tl.command('results.publish', properties, '');
+}
+
+export async function publishTestResults(testResultsFiles: string[], searchFolder: string, testRunner: TestRunner = 'NUnit') {
+
+    if (isNullOrWhitespace(searchFolder)) {
+        searchFolder = tl.getVariable('System.DefaultWorkingDirectory');
+    }
+
+    // Sending allowBrokenSymbolicLinks as true, so we don't want to throw error when symlinks are broken.
+    // And can continue with other files if there are any.
+    const findOptions = <tl.FindOptions>{
+        allowBrokenSymbolicLinks: true,
+        followSpecifiedSymbolicLink: true,
+        followSymbolicLinks: true
+    };
+
+    const matchingTestResultsFiles = tl.findMatch(searchFolder, testResultsFiles, findOptions);
```

Comment:
We have deterministic logic for where `TestResult.xml` is going to be stored. Can we add this logic later?